### PR TITLE
#5686-Added implementation for change and delete source case test

### PR DESCRIPTION
**Pull request by Mihai-Bota**
Added implementation for item 5686

used test for change source
added an extra change source step and check
added remove source case step

No allure report due to not working allure library on M1 macbook. Please check 15 times script running on your side.
On my side it worked well with 15 runs.

### CODE REVIEW

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/ChooseSourceCaseSteps.java [hunk 1]
```diff
@@ -52,25 +48,31 @@ public ChooseSourceCaseSteps(
...
-          webDriverHelpers.clickWhileOtherButtonIsDisplayed(
-              SOURCE_CASE_WINDOW_CONFIRM_BUTTON, CHANGE_CASE_BUTTON);
+          webDriverHelpers.waitUntilElementIsVisibleAndClickable(SOURCE_CASE_WINDOW_CONFIRM_BUTTON);
+          webDriverHelpers.clickOnWebElementBySelector(SOURCE_CASE_WINDOW_CONFIRM_BUTTON);
+          webDriverHelpers.waitUntilIdentifiedElementIsVisibleAndClickable(
+              SUCCESSFUL_CASE_CHANGE_POPUP);
         });
 
     When(
         "I click on the CHOOSE SOURCE CASE button",
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/ChooseSourceCaseSteps.java:58]: I know that is out of this PR, but can we change the name of the gherkin syntax to reflect this action?

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/ChooseSourceCaseSteps.java [hunk 2]
```diff
@@ -52,25 +48,31 @@ public ChooseSourceCaseSteps(
...
         () ->
             webDriverHelpers.clickWhileOtherButtonIsDisplayed(
-                CHOOSE_SOURCE_CASE_BUTTON, DISCARD_POPUP_YES_BUTTON));
+                CHOOSE_SOURCE_CASE_BUTTON, POPUP_YES_BUTTON));
 
     When(
         "I click yes on the DISCARD UNSAVED CHANGES popup",
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/ChooseSourceCaseSteps.java:64]: also here, can we add also the location of where this action happens?

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/ChooseSourceCaseSteps.java [hunk 3]
```diff
@@ -97,11 +99,36 @@ public ChooseSourceCaseSteps(
...
         });
+
+    When(
+        "I click on the CHANGE CASE button",
+        () ->
+            webDriverHelpers.clickWhileOtherButtonIsDisplayed(
+                CHANGE_CASE_BUTTON, POPUP_YES_BUTTON));
+
+    When(
+        "I click on the Remove Case CTA",
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/ChooseSourceCaseSteps.java:116]: CTA stands for control to action?

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/ChooseSourceCaseSteps.java [hunk 4]
```diff
@@ -97,11 +99,36 @@ public ChooseSourceCaseSteps(
...
+
+    Then(
+        "I check the CHOOSE SOURCE CASE BUTTON is displayed",
+        () ->
+            assertWithMessage("The expected element was not displayed")
+                .that(webDriverHelpers.isElementVisibleWithTimeout(CHOOSE_SOURCE_CASE_BUTTON, 1))
+                .isTrue());
+
+    When(
+        "I click yes on the CONFIRM REMOVAL popup",
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/ChooseSourceCaseSteps.java:129]: also, add the location of this removal popup

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/ChooseSourceCaseSteps.java (hunk 3 above)
Comment (Mihai-Bota) [reply sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/ChooseSourceCaseSteps.java:116]: call to action

**Review by alinapopaiustina — APPROVED** [review]

**Review by VGitVlad — APPROVED** [review]

File: sormas-e2e-tests/src/main/java/org/sormas/e2etests/pages/application/contacts/EditContactPage.java [hunk 5]
```diff
@@ -113,4 +113,7 @@
   public static final By CASE_CLASSIFICATION_LABEL =
       By.xpath(
           "//span[contains(text(), 'Case classification')]/ancestor::div[@class='v-caption']/following-sibling::div");
+  public static final By REMOVE_CASE_CTA_LINK = By.id("contactRemoveCase");
+  public static final By SUCCESSFUL_CASE_CHANGE_POPUP =
```
[the remark this answers is not in the document]
Comment (Mihai-Bota) [reply sormas-e2e-tests/src/main/java/org/sormas/e2etests/pages/application/contacts/EditContactPage.java]: reworked

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/ChooseSourceCaseSteps.java (hunk 1 above)
Comment (Mihai-Bota) [reply sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/ChooseSourceCaseSteps.java]: reworked

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/ChooseSourceCaseSteps.java (hunk 2 above)
Comment (Mihai-Bota) [reply sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/ChooseSourceCaseSteps.java]: reworked

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/ChooseSourceCaseSteps.java (hunk 4 above)
Comment (Mihai-Bota) [reply sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/contacts/ChooseSourceCaseSteps.java]: reworked

**Review by rdutu-vg — APPROVED** [review]

**Pull request merged**